worldedit-bukkit/adapters/adapter-1_17_1/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/fawe/v1_17_R1_2/PaperweightFaweAdapter.java

```diff
@@ -555,6 +571,164 @@ protected ServerLevel getServerLevel(final World world) {
         return ((CraftWorld) world).getHandle();
     }
 
+    @Override
```

Comment:
There is some similarity to the generateTree code here. Can we move it to the FaweAdapter too? I also think the Map<BlockPos, CraftBlockState> can be simplified to a list similar to how it's done in generateTree now.

Reply:
I was going to look at doing so yeah. It would be more complicated though as the method changes per version a couple of times

Reply:
Yeah I don't think that's very viable. We're not making any meaningful amount of abstraction as almost every line has a NMS class